scripts/python/Ne_from_SLiM_vcfs.py

```diff
@@ -71,32 +66,37 @@ def create_sfs_dict(inpath):
 def write_thetaNe_values(sfs_dict, outpath):
 
     # Open csv to write resutls
-    outfile = open(outpath, 'w+')
-
-    # Write header
-    outfile.write('N,bot,gen,theta_pi,theta_w,Ne_pi,Ne_w\n')
-
-    mu = float(1e-8)
-
-    # Interate through sfs dictionary and write to csv
-    for key in sfs_dict.keys():
-
-        # Get dict values
-        d = sfs_dict[key]
-
-        l = d.keys()
-        l = sorted(l)
-
-        # Iterate through nested dictionary values (generations)
-        for g in l:
-            pi = d[g].theta_pi()
-            watersons = d[g].theta_w()
-            Ne_pi = (pi / (4 * mu))
-            Ne_w = (watersons / (4 * mu))
-#             print(g, d[g].theta_pi(), d[g].theta_w(),Ne_pi,Ne_w)
-            N_and_Gen = key.split('-')
-            outfile.write('{0},{1},{2},{3},{4},{5},{6}\n'.format(N_and_Gen[0], N_and_Gen[1], g, pi,watersons, Ne_pi, Ne_w))
-    outfile.close()
+    with open(outpath, 'w+') as f:
+
+        # Write header
+        f.write('N,bot,gen,theta_pi,theta_w,Ne_pi,Ne_w\n')
+
+        mu = float(1e-8)
+
+        # Interate through sfs dictionary and write to csv
+        for key in sfs_dict.keys():
+
+            # Get dict values, which are encoded as <pop size>-<bottleneck>
+            size_bot = sfs_dict[key]
+
+            # Get generations as list
+            generations = sorted(size_bot.keys())
+
+            # Iterate through nested dictionary values (generations)
+            for gen in generations:
+                pi = size_bot[gen].theta_pi()
+                watersons = size_bot[gen].theta_w()
+                Ne_pi = round((pi / (4 * mu)), 3)
+                Ne_w = round((watersons / (4 * mu)), 3)
+                pop_size = key.split('-')[0]
+                bottleneck = key.split('-')[1]
+                f.write('{0},{1},{2},{3},{4},{5},{6}\n'.format(pop_size,
```

Comment:
Also inconsequential but I find `','.join([pop_size, bottleneck, gen, pi, watersons, Ne_pi, Ne_w]) + '\n'` to be a bit more readable for this sort of thing

Reply:
Decided to use `csv` module instead so that the parameters could be written as a simple list, which avoids the use of both `','.join()` and `.format()`